samples/nrf9160/modem_shell/src/sms/sms_shell.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <shell/shell.h>
+#include <assert.h>
+#include <strings.h>
+#include <stdio.h>
+#include <getopt.h>
+
+#include "sms.h"
+#include "mosh_defines.h"
+
+/* Maximum length of the message data that can be specified with -m option */
+#define SMS_MAX_MESSAGE_LEN CONFIG_SHELL_CMD_BUFF_SIZE
+
+enum sms_command {
+	SMS_CMD_REGISTER = 0,
+	SMS_CMD_UNREGISTER,
+	SMS_CMD_SEND,
+	SMS_CMD_RECV,
+	SMS_CMD_HELP
+};
+
+extern const struct shell *shell_global;
+
+const char sms_usage_str[] =
+	"Usage: sms <command> [options]\n"
+	"\n"
+	"<command> is one of the following:\n"
+	"  send:    Send SMS message. Also registers SMS service if required.\n"
+	"           Mandatory options: -m, -n\n"
+	"  recv:    Record number of received SMS messages.\n"
+	"  reg:     Register SMS service to be able to receive messages.\n"
+	"  unreg:   Unregister SMS service after which messages cannot be received.\n"
+	"\n"
+	"Options for 'send' command:\n"
+	"  -m, --message, <str> Text to be sent.\n"
+	"  -n, --number, <str>  Number in international format including country number.\n"
+	"                       Leading '+' can be present or left out.\n"
+	"\n"
+	"Options for 'recv' command:\n"
+	"  -r, --start,         Reset SMS counter to zero\n"
+	"\n"
+	"Options for 'help' command:\n"
+	"  -v, --verbose,       Show examples\n";
+
+const char sms_usage_example_str[] =
+	"Examples:\n"
+	"\n"
+	"TODO\n";
```

Comment:
I'm currently thinking to remove all examples from the usage and include those into the documentation. This would also reduce the memory footprint.
What do you think?

Reply:
Good idea.

Reply:
Done. Removed examples from sock, sms and AT modules and they are now in the documentation side.